### PR TITLE
Keyvault track 2 vault client readme

### PR DESCRIPTION
**Pull request by samvaity**
Track 2  Vault client readme
Closes: #5662

cc: @Azure/adp-keyvault

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/keyvault/azure-security-keyvault/README.md [hunk 1]
```diff
@@ -1,13 +1,116 @@
...
+Use the Key Vault client library to:
 
-# Examples
+* Securely store and control access to tokens, passwords, certificates, API keys, and other secrets.
+* Create and control encryption keys that encrypt your data.
+* Provision, manage, and deploy public and private Secure Sockets Layer/Transport Layer Security (SSL/TLS) certificates for use with Azure and your internal connected resources.
+* Use either software or FIPS 140-2 Level 2 validated HSMs to help protect secrets and keys.
 
-# Troubleshooting
+[Source code](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-security-keyvault) | [Package (Pypi)](https://pypi.org/project/azure-security-keyvault/) | [API reference documentation](https://docs.microsoft.com/python/api/azure-security-keyvault) | [Product documentation](https://docs.microsoft.com/en-gb/azure/key-vault/)
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault/README.md]: can this potentially change once we decide to package it separately?

File: sdk/keyvault/azure-security-keyvault/README.md [hunk 2]
```diff
@@ -1,13 +1,116 @@
...
+`$ pip install azure-security-keyvault
+`
+
+### Prerequisites
+* An [Azure subscription](https://azure.microsoft.com/free/).
+* Python 2.7, 3.4 or later.
+* An existing Key Vault. You can create this by following the instructions on [this article](https://docs.microsoft.com/en-gb/azure/key-vault/quick-create-portal).
+
+### Authenticate the client
+In order to interact with the Key Vault service, you'll need to create an instance of the [KeyVaultClient](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-security-keyvault/azure/security/keyvault) class. For this, you will need a vault url, azure credentials, client id, client secret, tenant id and resource url.
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault/README.md]: this is going to change, I am guessing?
How should this look like?

File: sdk/keyvault/azure-security-keyvault/README.md [hunk 3]
```diff
@@ -1,13 +1,116 @@
...
+logging.basicConfig(level=logging.DEBUG)
+```
+Http request and response details are printed to stdout with this logging config.
+
+## Next steps
+* Set and retrieve a secret [TODO: add link]
+* Set and retrieve a key [TODO: add link]
+
+###  Documentation
+Reference documentation is available at docs.microsoft.com/python/api/azure-security-keyvault
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault/README.md]: it is expected to have the new documentation at this location, but it does not exist currently?

**Comment by chlowell** [comment]
Closing this because we aren't releasing `VaultClient` or a metapackage that would benefit from a general Key Vault readme.

**Comment by kaerm** [comment]
Even if there is no metapackage there should be a general key vault service level readme with simple links to specific key vault package READMEs and one line explanations on when to use them - otherwise navigating the repo gets complicated and confusing

**Pull request closed without merging**